packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -503,6 +506,20 @@ function createAssumeRolePolicy(principal: IPrincipal, externalIds: string[]) {
   return actualDoc;
 }
 
+function validateRolePath(path?: string) {
+  if (path === undefined || Token.isUnresolved(path)) {
+    return;
+  }
+
+  const validRolePath = /^(\/|\/[\u0021-\u007F]+\/)$/;
+
+  if (path.length == 0 || path.length > 512) {
+    throw new Error(`role path is set to ${path}, but the length must be >= 1 and <= 512.`);
+  } else if (!validRolePath.test(path)) {
+    throw new Error(`role path is set to ${path}, but must match ${validRolePath.toString()}.`);
```

Comment:
I'm not totally sure what the best error message is here but providing the regex itself here is not very user friendly. Perhaps you can describe the pattern needed instead of just using the regex. also, please capitalize the beginning of the sentence in the error message.

Reply:
I've added a bit more explanation. Updated in ea3a32439144a1f3597dd0bf14e41f3fe16973b9